File: python/robot_return_to_origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        vertical, horizontal = 0, 0
        for move in moves:
            if move == 'U': vertical += 1
            elif move == 'D': vertical -= 1
            elif move == 'R': horizontal += 1
            elif move == 'L': horizontal -= 1
        return vertical == 0 and horizontal == 0

File: src/RobotReturnToOrigin.java
public class RobotReturnToOrigin {
    public boolean judgeCircle(String moves) {
        int vertical = 0, horizontal = 0;
        for (int index = 0 ; index < moves.length() ; index++) {
            switch (moves.charAt(index)) {
                case 'U': vertical++; break;
                case 'R': horizontal++; break;
                case 'D': vertical--; break;
                case 'L': horizontal--; break;
            }
        }
        return vertical == 0 && horizontal == 0;
    }
}
